packages/unchained-client/src/utxo/parser/thorchain.ts

```diff
@@ -0,0 +1,249 @@
+import type { Axios } from 'axios'
+import axios from 'axios'
+
+import { type BaseTxMetadata, Dex, TradeType } from '../../types'
+import type { SubParser, Tx, TxSpecific } from '../parser'
+
+export type LiquidityType = 'Savers' | 'LP'
+export type SwapType = 'Standard' | 'Streaming'
+
+interface Liquidity {
+  type: LiquidityType
+}
+
+interface Swap {
+  type: SwapType
+}
+
+export interface TxMetadata extends BaseTxMetadata {
+  parser: 'thorchain'
+  memo: string
+  liquidity?: Liquidity
+  swap?: Swap
+}
+
+interface Coin {
+  amount: string
+  asset: string
+}
+
+interface InOut {
+  txID: string
+  address: string
+  coins: Coin[]
+  height?: string
+}
+
+interface Metadata {
+  addLiquidity?: {
+    liquidityUnits: string
+  }
+  refund?: {
+    affiliateAddress: string
+    affiliateFee: string
+    memo: string
+    networkFees: Coin[]
+    reason: string
+  }
+  swap?: {
+    affiliateAddress: string
+    affiliateFee: string
+    isStreamingSwap: boolean
+    liquidityFee: string
+    networkFees: Coin[]
+    swapSlip: string
+    swapTarget: string
+    memo: string
+  }
+  withdraw?: {
+    asymmetry: string
+    basisPoints: string
+    impermanentLossProtection: string
+    liquidityUnits: string
+    memo: string
+    networkFees: Coin[]
+  }
+}
+
+interface Actions {
+  date: string
+  height: string
+  in: InOut[]
+  out: InOut[]
+  metadata?: Metadata
+  pools: string[]
+  status: 'success' | 'pending'
+  type: 'swap' | 'addLiquidity' | 'withdraw' | 'donate' | 'refund' | 'switch'
+}
+
+export interface ActionsResponse {
+  actions: Actions[]
+  count: string
+  meta: {
+    nextPageToken: string
+    prevPageToken: string
+  }
+}
+
+interface ExtraMetadata {
+  method: string
+  liquidity?: Liquidity
+  swap?: Swap
+}
+
+const getLiquidityType = (pool: string): LiquidityType => (pool.includes('/') ? 'Savers' : 'LP')
```

Comment:
🧠

Reply:
As discussed, savers pools use the `/` delimiter while regular pools use `.` e.g

- Savers

`ETH/USDC-0XA0B86991C6218B36C1D19D4A2E9EB0CE3606EB48` https://viewblock.io/thorchain/tx/3056EEA5D6336CE7CDDDD30B74E2F9E0BB170428982052B2C49AF517E2A96FDF

 - Actual pool
 
 `ETH.USDC-0XA0B86991C6218B36C1D19D4A2E9EB0CE3606EB48` 
 https://viewblock.io/thorchain/tx/2050C3A891B3D3C7D39255E96ED4DA816057011B382FC1F6099E6B8958058268